inference-engine/src/transformations/src/transformations/convert_precision.cpp

```diff
@@ -297,8 +297,40 @@ bool extend_select_type(std::shared_ptr<ngraph::Node> & node, ngraph::element::T
     return false;
 }
 
+template <typename src_type, typename dst_type>
+inline dst_type convert_value(src_type val) {
+    if (val > std::numeric_limits<dst_type>::max()) {
+        return std::numeric_limits<dst_type>::max();
+    } else if (val < std::numeric_limits<dst_type>::lowest()) {
+        return std::numeric_limits<dst_type>::lowest();
+    }
+    return static_cast<dst_type>(val);
+}
+
+// We need to treat U64->I32 and U32->I32 as a separate case, because of C++'s implicit promotion from signed to unsigned.
+// So before comparing value to std::numeric_limits<int32_t>::lowest(), we need to cast it to the same width signed type.
+template <>
+inline int32_t convert_value<uint64_t, int32_t>(uint64_t val) {
+    if (val > std::numeric_limits<int32_t>::max()) {
+        return std::numeric_limits<int32_t>::max();
+    } else if (static_cast<int64_t>(val) < std::numeric_limits<int32_t>::lowest()) {
+        return std::numeric_limits<int32_t>::lowest();
```

Comment:
I think it is not correct. We shouldn't cast `uint64_t` to `int64_t`. From big positive number we can get a negative int. I think we should round it to `std::numeric_limits<int32_t>::max()`

Reply:
In case of big positive number we'll never get there, because in that case we take branch at line 314 before

Reply:
In this case this check looks redundant... Because min of unsigned data type is 0.

Reply:
right. I'll remove it.